airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3BaseChecks.java

```diff
@@ -77,6 +77,17 @@ public static void testMultipartUpload(final AmazonS3 s3Client, final String buc
     LOGGER.info("Finished verification for multipart upload mode");
   }
 
+  /**
+   * Checks that S3 custom endpoint uses a variant that only uses HTTPS
+   * <p>https://docs.aws.amazon.com/general/latest/gr/s3.html</p>
+   * @param endpoint URL string representing an accessible S3 bucket
+   */
+  public static void testCustomEndpointSecured(final String endpoint) {
+    if (!endpoint.contains("s3-accesspoint")) {
```

Comment:
I think this might be overzealous and also not catch all problematic endpoints? It's possible to self-host an S3-compatible API, which is mostly what this config is used for. And that's considered secure, assuming the connection uses HTTPS. E.g. I might have `https://mystorage.edgao.example.com`.

But I could also (hypothetically) host my API at `http://s3-accesspoint.edgao.example.com` - which passes this check, but is not secured.

Reply:
also, I think we should only be enforcing this check in cloud. I.e. OSS users might want to use unsecured connections intentionally, and we should allow that to happen. (maybe they're hosting airbyte and S3 within their own network, and are OK with having unsecured traffic within their network)

also also, where is this method actually invoked? I'm guessing you have a local unpushed commit or something?

Reply:
Updated the check to validate if `https://` is contained in the endpoint. Currently looking into using `AdaptiveDestinationRunner` on how to get this feature only within cloud and _not_ within OSS users

Also added the method invocation within `BaseS3Destination` since it accidentally got cleared due to the overhaul of S3 heirarchy 🤦‍♂️